core/env/methods_spec.rb

```diff
@@ -0,0 +1,39 @@
+require_relative '../../spec_helper'
+
+ruby_version_is "2.7" do
+  describe "ENV.methods" do
+    it "includes new methods" do
+      [:merge].each do |new_method|
+        ENV.methods.should include(new_method)
+      end
+    end
+  end
+end
+
+ruby_version_is "2.6" do
+  describe "ENV.methods" do
+    it "includes new methods" do
+      [:filter, :filter!, :slice].each do |new_method|
+       ENV.methods.should include(new_method)
+      end
+    end
+  end
+end
+
+ruby_version_is "2.5" do
+  describe "ENV.class" do
+    ENV.should be_an_instance_of(Object)
+  end
+
+  describe "ENV.methods" do
+    it "returns an array of methods" do
+      methods = ENV.methods false
+      expected_methods = [:[], :[]=, :assoc, :clear, :delete, :delete_if, :each, :each_key, :each_pair, :each_value, :empty?, :fetch, :has_key?, :has_value?, :include?, :index, :inspect, :invert, :keep_if, :key, :key?, :keys, :length, :member?, :rassoc, :rehash, :reject, :reject!, :replace, :select, :select!, :shift, :size, :store, :to_a, :to_h, :to_hash, :to_s, :update, :value?, :values, :values_at]
```

Comment:
Listing all methods like this is fairly unusual for specs, I am not sure it's a good idea here.
Could you extract that commit in a separate PR?

In general, each method should have specs and passing those already guarantees all methods are implemented.
Having extra methods should probably not be forbidden.

Reply:
Sure, will separate.  I was thinking that this would prompt us to spec a new method.